fili-core/src/main/java/com/yahoo/bard/webservice/table/BasePhysicalTable.java

```diff
@@ -3,23 +3,25 @@
 package com.yahoo.bard.webservice.table;
 
 import com.yahoo.bard.webservice.data.config.names.TableName;
-import com.yahoo.bard.webservice.data.dimension.DimensionDictionary;
 import com.yahoo.bard.webservice.data.time.ZonedTimeGrain;
-import com.yahoo.bard.webservice.metadata.SegmentMetadata;
+import com.yahoo.bard.webservice.druid.model.query.Granularity;
 import com.yahoo.bard.webservice.table.availability.Availability;
-import com.yahoo.bard.webservice.table.availability.ImmutableAvailability;
+import com.yahoo.bard.webservice.table.resolver.DataSourceConstraint;
 import com.yahoo.bard.webservice.util.IntervalUtils;
+import com.yahoo.bard.webservice.util.SimplifiedIntervalList;
 
 import org.joda.time.DateTime;
+import org.joda.time.Interval;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.util.Map;
+import java.util.Set;
 
 import javax.validation.constraints.NotNull;
 
 /**
- * Base Physical Table implements common PhysicalTable capabilities.
+ * Base implementation of physical table that are shared across various types of physical tables.
  */
 public abstract class BasePhysicalTable implements PhysicalTable {
```

Comment:
Right now, `BasePhysicalTable` is calling `containsLogicalName` just to invoke the log, so that's not really a change. When possible, we should push complexity lower in the stack.

For reference, here the current code in `BasePhysicalTable`:

```java
/**
 * Translate a logical name into a physical column name. If no translation exists (i.e. they are the same),
 * then the logical name is returned.
 * <p>
 * NOTE: This defaulting behavior <em>WILL BE REMOVED</em> in future releases.
 * <p>
 * The defaulting behavior shouldn't be hit for Dimensions that are serialized via the default serializer and are
 * not properly configured with a logical-to-physical name mapping. Dimensions that are not "normal" dimensions,
 * such as dimensions used for DimensionSpecs in queries to do mapping from fact-level dimensions to something else,
 * should likely use their own serialization strategy so as to not hit this defaulting behavior.
 *
 * @param logicalName  Logical name to lookup in physical table
 *
 * @return Translated logicalName if applicable
 */
public String getPhysicalColumnName(String logicalName) {
    if (!schema.containsLogicalName(logicalName)) {
        LOG.warn(
                "No mapping found for logical name '{}' to physical name on table '{}'. Will use logical name as " +
                        "physical name. This is unexpected and should not happen for properly configured " +
                        "dimensions.",
                logicalName,
                getName()
        );
    }
    return schema.getPhysicalColumnName(logicalName);
}
```

and the code in `PhysicalTableSchema`:

```java
/**
 * Translate a logical name into a physical column name. If no translation exists (i.e. they are the same),
 * then the logical name is returned.
 * <p>
 * NOTE: This defaulting behavior <em>WILL BE REMOVED</em> in future releases.
 * <p>
 * The defaulting behavior shouldn't be hit for Dimensions that are serialized via the default serializer and are
 * not properly configured with a logical-to-physical name mapping. Dimensions that are not "normal" dimensions,
 * such as dimensions used for DimensionSpecs in queries to do mapping from fact-level dimensions to something else,
 * should likely use their own serialization strategy so as to not hit this defaulting behavior.
 *
 * @param logicalName  Logical name to lookup in physical table
 *
 * @return Translated logicalName if applicable
 */
public String getPhysicalColumnName(String logicalName) {
    return logicalToPhysicalColumnNames.getOrDefault(logicalName, logicalName);
}
```